homeassistant/loader.py

```diff
@@ -993,10 +994,7 @@ def _load_file(
             comp_or_platform
         ]
 
-    if (cache := hass.data.get(DATA_COMPONENTS)) is None:
-        if not _async_mount_config_dir(hass):
-            return None
```

Comment:
Do we want to keep this? We could remember if `_async_mount_config_dir` failed and bail out early here if it did?